// custom-logging-module/dist/models/ColorsMaps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LightColorMapping = exports.DarkColorMapping = void 0;
const TerminalColors_1 = require("../types/TerminalColors");
exports.DarkColorMapping = {
    Black: TerminalColors_1.TerminalDarkColors.Black,
    Red: TerminalColors_1.TerminalDarkColors.Red,
    Green: TerminalColors_1.TerminalDarkColors.Green,
    Yellow: TerminalColors_1.TerminalDarkColors.Yellow,
    Blue: TerminalColors_1.TerminalDarkColors.Blue,
    Magenta: TerminalColors_1.TerminalDarkColors.Magenta,
    Cyan: TerminalColors_1.TerminalDarkColors.Cyan,
    White: TerminalColors_1.TerminalDarkColors.White,
    Random: TerminalColors_1.TerminalDarkColors.Black // Par défaut
};
exports.LightColorMapping = {
    Black: TerminalColors_1.TerminalLightColors.Black,
    Red: TerminalColors_1.TerminalLightColors.Red,
    Green: TerminalColors_1.TerminalLightColors.Green,
    Yellow: TerminalColors_1.TerminalLightColors.Yellow,
    Blue: TerminalColors_1.TerminalLightColors.Blue,
    Magenta: TerminalColors_1.TerminalLightColors.Magenta,
    Cyan: TerminalColors_1.TerminalLightColors.Cyan,
    White: TerminalColors_1.TerminalLightColors.White,
    Random: TerminalColors_1.TerminalLightColors.White // Par défaut
};
