// custom-logging-module/dist/types/TerminalColors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ColorChoice = exports.TerminalLightColors = exports.TerminalDarkColors = exports.TerminalColors = void 0;
var TerminalColors;
(function (TerminalColors) {
    TerminalColors["Dark_Black"] = "\u001B[30m";
    TerminalColors["Dark_Red"] = "\u001B[31m";
    TerminalColors["Dark_Green"] = "\u001B[32m";
    TerminalColors["Dark_Yellow"] = "\u001B[33m";
    TerminalColors["Dark_Blue"] = "\u001B[34m";
    TerminalColors["Dark_Magenta"] = "\u001B[35m";
    TerminalColors["Dark_Cyan"] = "\u001B[36m";
    TerminalColors["Dark_White"] = "\u001B[37m";
    TerminalColors["LightBlack"] = "\u001B[90m";
    TerminalColors["LightRed"] = "\u001B[91m";
    TerminalColors["LightGreen"] = "\u001B[92m";
    TerminalColors["LightYellow"] = "\u001B[93m";
    TerminalColors["LightBlue"] = "\u001B[94m";
    TerminalColors["LightMagenta"] = "\u001B[95m";
    TerminalColors["LightCyan"] = "\u001B[96m";
    TerminalColors["LightWhite"] = "\u001B[97m";
})(TerminalColors || (exports.TerminalColors = TerminalColors = {}));
var TerminalDarkColors;
(function (TerminalDarkColors) {
    TerminalDarkColors["Black"] = "\u001B[30m";
    TerminalDarkColors["Red"] = "\u001B[31m";
    TerminalDarkColors["Green"] = "\u001B[32m";
    TerminalDarkColors["Yellow"] = "\u001B[33m";
    TerminalDarkColors["Blue"] = "\u001B[34m";
    TerminalDarkColors["Magenta"] = "\u001B[35m";
    TerminalDarkColors["Cyan"] = "\u001B[36m";
    TerminalDarkColors["White"] = "\u001B[37m";
})(TerminalDarkColors || (exports.TerminalDarkColors = TerminalDarkColors = {}));
var TerminalLightColors;
(function (TerminalLightColors) {
    TerminalLightColors["Black"] = "\u001B[90m";
    TerminalLightColors["Red"] = "\u001B[91m";
    TerminalLightColors["Green"] = "\u001B[92m";
    TerminalLightColors["Yellow"] = "\u001B[93m";
    TerminalLightColors["Blue"] = "\u001B[94m";
    TerminalLightColors["Magenta"] = "\u001B[95m";
    TerminalLightColors["Cyan"] = "\u001B[96m";
    TerminalLightColors["White"] = "\u001B[97m";
})(TerminalLightColors || (exports.TerminalLightColors = TerminalLightColors = {}));
var ColorChoice;
(function (ColorChoice) {
    ColorChoice["Black"] = "Black";
    ColorChoice["Red"] = "Red";
    ColorChoice["Green"] = "Green";
    ColorChoice["Yellow"] = "Yellow";
    ColorChoice["Blue"] = "Blue";
    ColorChoice["Magenta"] = "Magenta";
    ColorChoice["Cyan"] = "Cyan";
    ColorChoice["White"] = "White";
    ColorChoice["Random"] = "Random";
})(ColorChoice || (exports.ColorChoice = ColorChoice = {}));
